packages/collections/map.pony

```diff
@@ -170,14 +183,13 @@ class HashMap[K, V, H: HashFunction[K] val]
 
         if (_size * 4) > (_array.size() * 3) then
           _resize(_array.size() * 2)
-          return this(key')?
         end
       end
 
-      _array(i)? as (_, V)
+      return _array(i)? as (_, V)
     else
       // This is unreachable, since index will never be out-of-bounds.
-      error
+      consume value''
```

Comment:
Is this `consume` necessary? I'd think that it shouldn't be, but maybe it is...

What error do you see if you remove it?